crates/query-engine/translation/src/translation/query/fields.rs

```diff
@@ -283,29 +283,80 @@ fn translate_nested_field(
         }
     }?;
 
+    // The recursive call to the next layer of fields
+    let (nested_field_table_reference, nested_field_binding_alias) = {
+        match &current_table.source {
+            TableSource::Collection(collection_name) => {
+                let source = TableSource::NestedField {
+                    collection_name: collection_name.clone(),
+                    type_name: nested_field_type_name,
+                    field_path: FieldPath(vec![current_column_name.clone()]),
+                };
+                let nested_field_binding_alias = state.make_table_alias(source.name_for_alias());
+                (
+                    TableSourceAndReference {
+                        source,
+                        reference: sql::ast::TableReference::AliasedTable(
+                            nested_field_binding_alias.clone(),
+                        ),
+                    },
+                    nested_field_binding_alias,
+                )
+            }
+            TableSource::NestedField {
+                collection_name,
+                type_name: _,
+                field_path,
+            } => {
+                let mut field_path = field_path.0.clone();
+                field_path.push(current_column_name.clone());
+                let source = TableSource::NestedField {
+                    collection_name: collection_name.clone(),
+                    type_name: nested_field_type_name,
+                    field_path: FieldPath(field_path),
```

Comment:
It looks like this lets us build up multiple nested field lookups in one go. Are there examples of this in tests?

Reply:
The `field_path` and `collection_name` are only used for naming the alias, so even if this isn't perfect, it wouldn't effect the query.

Reply:
I think if they're only used as a convenience we should consider making that more obvious in the code. There is a lot of stuff that has happen, so the more we can direct the reader's attention the better.

Reply:
Yeah I understand. I did try to make this explicit in the type's comments:
https://github.com/hasura/ndc-postgres/pull/564/files#diff-d24352dcf20dd7c6e0b5e4f0c7dec878e49c311b976280b19008620bb936bb2eR80

But I guess there is a discoverability issue here.